test/lib.sh

```diff
@@ -59,3 +59,59 @@ EOF
 hr() {
     echo "========================="
 }
+
+diff() {
+	for file in `ls` ; do 
+		if [ -d $file ] ; then 
+			pushd $file  > /dev/null
+			diff $1/$file 
+			popd  > /dev/null
+		elif [ -f $file ] ; then
+			cmp $file $1/$file
+			if [ ! $? = 0 ] ; then
+				rc=`expr $rc+1`
+			fi
+		fi
+	done
+	return 0
+}
+
+
+compare() {
+	if [ $1 = "_data" ] ; then
+		if [ ! -f $repo/boilerplate/_data/last_boilerplate_commit ] ; then
+			rc=`expr $rc+1`
```

Comment:
Need to initialize `rc`. Unless you're counting on it coming down from the caller -- which IMO is not a good look.

Also, just set it to a hardcoded value rather than incrementing it, per [above comment](https://github.com/openshift/boilerplate/pull/11/files#diff-e1b2b21859e872e9af54745ed0a1b5c4R76). I know in this case it can only be hit once, but good habits.

Reply:
If I am not mistaken, the `declare -i rc` (line 107) is initialising it to 0. The objective is to use to rc to return how many checks failed basically, hence the init in the check_method.

Reply:
Okay, right, so to expand on my "not a good look" comment, it's not good practice for the two functions to be so tightly coupled. In particular, if I wanted to use `compare` from somewhere else, I would have to be sure to set up `rc` beforehand. That's weird -- and also not really necessary.

If you want to count failures from `compare`, you can increment a variable local to the caller _from_ the caller. (But we still shouldn't use that count as a return code, for reasons stated.)

Reply:
I implemented a dedicated function for init/increment/check the counter actually.